### ISSUE TITLE
[react-snap-carousel]: Unsupported scroll margin value, expected <length> value, received 

### ISSUE THREAD

**Issue by KasperAndersson** [richardscarrott/react-snap-carousel#25]
Using `react-snap-carousel` in older browser, from our issue tracker seems to be below iOS 15: `[react-snap-carousel]: Unsupported scroll margin value, expected <length> value, received ` is thrown, making our nextjs app go `Application error: a client-side exception has occurred (see the browser console for more information)`. 

I can't quite figure out, what in the library that's not supported by iOS 14 (both scroll-padding and scroll-margin should be supported), would it make sense to either be able to pass a fallback `value` or bypass the lib throwing errors?

**Update**
Patched like this, and now it works in iOS 14

```javascript
diff --git a/node_modules/react-snap-carousel/dist/use-snap-carousel.js b/node_modules/react-snap-carousel/dist/use-snap-carousel.js
index 48c97d5..2c3162e 100644
--- a/node_modules/react-snap-carousel/dist/use-snap-carousel.js
+++ b/node_modules/react-snap-carousel/dist/use-snap-carousel.js
@@ -212,7 +212,7 @@ var _getOffsetRect = (el) => {
 };
 var getScrollPaddingUsedValue = (el, pos) => {
   const style = window.getComputedStyle(el);
-  const scrollPadding = style.getPropertyValue(`scroll-padding-${pos}`);
+  const scrollPadding = style.getPropertyValue(`scroll-padding-${pos}`) || "0px";
   if (scrollPadding === "auto") {
     return 0;
   }
@@ -231,7 +231,7 @@ var getScrollPaddingUsedValue = (el, pos) => {
 };
 var getScrollMarginUsedValue = (el, pos) => {
   const style = window.getComputedStyle(el);
-  const scrollMargin = style.getPropertyValue(`scroll-margin-${pos}`);
+  const scrollMargin = style.getPropertyValue(`scroll-margin-${pos}`) || "0px";
   const invalidMsg = `Unsupported scroll margin value, expected <length> value, received ${scrollMargin}`;
   assert(scrollMargin.endsWith("px"), invalidMsg);
   const value = parseInt(scrollMargin);
diff --git a/node_modules/react-snap-carousel/dist/use-snap-carousel.mjs b/node_modules/react-snap-carousel/dist/use-snap-carousel.mjs
index 4f87451..4bd9f61 100644
--- a/node_modules/react-snap-carousel/dist/use-snap-carousel.mjs
+++ b/node_modules/react-snap-carousel/dist/use-snap-carousel.mjs
@@ -188,7 +188,7 @@ var _getOffsetRect = (el) => {
 };
 var getScrollPaddingUsedValue = (el, pos) => {
   const style = window.getComputedStyle(el);
-  const scrollPadding = style.getPropertyValue(`scroll-padding-${pos}`);
+  const scrollPadding = style.getPropertyValue(`scroll-padding-${pos}`) || "0px";
   if (scrollPadding === "auto") {
     return 0;
   }
@@ -207,7 +207,7 @@ var getScrollPaddingUsedValue = (el, pos) => {
 };
 var getScrollMarginUsedValue = (el, pos) => {
   const style = window.getComputedStyle(el);
-  const scrollMargin = style.getPropertyValue(`scroll-margin-${pos}`);
+  const scrollMargin = style.getPropertyValue(`scroll-margin-${pos}`) || "0px";
   const invalidMsg = `Unsupported scroll margin value, expected <length> value, received ${scrollMargin}`;
   assert(scrollMargin.endsWith("px"), invalidMsg);
   const value = parseInt(scrollMargin);

```

**Comment by richardscarrott** [comment]
@KasperAndersson thanks reporting this -- I'd seen this in some error logs on a site I was working on a while back but didn't have the time to investigate.

I've pushed up your fix here https://github.com/richardscarrott/react-snap-carousel/pull/26

**Comment by richardscarrott** [comment]
@KasperAndersson I've published a new beta version with your fix which you can try out using `npm add react-snap-carousel@0.4.1-next.0`.

I'm going to ship it to production on a site I'm working on and if it's all good will release it under the `latest` tag as `0.4.1` sometime next week 👍

**Comment by richardscarrott** [comment]
Published as 0.4.1
